### PR TITLE
refactor: ensure task manager is retry-able and test-able

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

By passing the tasks to the manager at construction time, and not immediately starting it, we can improve how it will be handled in the exporter's open phase if that gets retried, and make sure it is retried in a safe way (e.g. do not resubmit the same task). That way, we ensure there is not the same task running twice which could cause concurrency issues.

Additionally, this will make it easier to unit and integration test, because we can now control when and how the tasks are executed, and have better introspection into the background task manager.

## Related issues

related to #24095

### CODE REVIEW

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/BackgroundTaskManager.java [hunk 1]
```diff
@@ -94,21 +117,19 @@ public static BackgroundTaskManager create(
...
-      final var batchOperationJob =
-          createBatchOperationJob(metrics, logger, resourceProvider, repository, executor);
-      archiver.start(config.getArchiver(), processInstanceJob, batchOperationJob);
-    } else {
-      archiver.start(config.getArchiver(), processInstanceJob);
+      tasks.add(createBatchOperationJob(metrics, logger, resourceProvider, repository, executor));
     }
 
-    return archiver;
+    executor.setCorePoolSize(Math.min(tasks.size(), MAX_BACKGROUND_THREADS));
```
Comment (rodrigo-lourenco-lopes) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/BackgroundTaskManager.java:130]: Nice work around to use a list of tasks 🙌

**Review by rodrigo-lourenco-lopes — APPROVED** [review]
Thanks for the changes and also the refactoring, looks better now 🚀

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/BackgroundTaskManager.java (hunk 1 above)
Comment (npepinpe) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/BackgroundTaskManager.java]: I updated it to actually just set the max size, and keep core size to 0. Threads time out after a minute, meaning if they have nothing to do for a minute, they will shut down.

So this means when we only have one job, it should pretty much keep the thread alive all the time. When we have two jobs, if there isn't much to do, the jobs will be interleaved, and we should still only use one thread :+1: If we have more work and more jobs, then we'll have more threads.

Comment (npepinpe) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/BackgroundTaskManager.java]: We could consider enforcing always having at minimum 1 thread anyway :shrug:

**Pull request merged**